pkg/kfapp/ossm/types.go

```diff
@@ -26,25 +22,55 @@ type manifest struct {
 	template,
 	patch,
 	processed bool
+	templateDir string
+}
+
+// In order to process the templates, we need to create a tmp directory
+// to store the files. This is because embedded files are read only.
+const baseOutputDir = "/tmp/ossm-installer/"
+
+func ensureDirExists(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err = os.MkdirAll(dir, 0755)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (m *manifest) targetPath() string {
-	return fmt.Sprintf("%s%s", m.path[:len(m.path)-len(filepath.Ext(m.path))], ".yaml")
+func (m *manifest) targetPath() (string, error) {
+	fullDir := filepath.Join(m.templateDir, filepath.Dir(m.path))
+	if err := ensureDirExists(fullDir); err != nil {
+		return "", err
+	}
+
+	fileName := filepath.Base(m.path)
+	fileNameWithoutExt := fileName[:len(fileName)-len(filepath.Ext(fileName))]
+	return filepath.Join(fullDir, fileNameWithoutExt+".yaml"), nil
 }
 
 func (m *manifest) processTemplate(data interface{}) error {
 	if !m.template {
 		return nil
 	}
-	f, err := os.Create(m.targetPath())
+	// Create yaml file in the regular filesystem
+	path, err := m.targetPath()
 	if err != nil {
+		log.Error(err, "Failed to generate target path")
+		return err
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		log.Error(err, "Failed to create file")
 		return err
 	}
 
 	tmpl := template.New(m.name).
 		Funcs(template.FuncMap{"ReplaceChar": ReplaceChar})
 
-	tmpl, err = tmpl.ParseFiles(m.path)
+	// Parse template from .tmpl file
+	tmpl, err = tmpl.ParseFiles(strings.Replace(path, ".yaml", ".tmpl", 1))
```

Comment:
I wonder why we need it. Originally the idea was that `m.path` holds the actual file to process (so in this case that would be `.tmpl`) but the `targetPath` has it swapped to `.yaml` on the fly. If the logic is preserved you don't need to do `.Replace` if you use `m.path`

Reply:
Yes, removed [22c5bd0](https://github.com/maistra/opendatahub-operator/pull/5/commits/22c5bd064ae796de82cc573fed39882477baab56)